core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -161,6 +162,25 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     this
   }
 
+  private final val avroNamespace = "avro.schema."
+
+  /**
+   * Use Kryo serialization and register the given set of Avro schemas so that the generic
+   * record serializer can decrease network IO
+   */
+  def registerAvroSchemas(schemas: Schema*): SparkConf = {
+    schemas.foldLeft(this) { (conf, schema) =>
+      conf.set(avroNamespace + SchemaNormalization.parsingFingerprint64(schema), schema.toString)
+    }
+  }
+
+  /** Gets all the avro schemas in the configuration used in the generic Avro record serializer */
+  def getAvroSchema: Map[Long, String] = {
```

Comment:
The keys are longs, which represent the a unique ID of the schema and the values are the string representation of the schema.